src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -673,3 +678,40 @@ int uORB::DeviceNode::update_queue_size(unsigned int queue_size)
 	_queue_size = queue_size;
 	return PX4_OK;
 }
+
+bool
+uORB::DeviceNode::register_work_item(px4::WorkItem *item)
+{
+
+	if (item != nullptr) {
+		ATOMIC_ENTER;
+
+		// prevent duplicate registrations
+		for (auto existing_item : _registered_work_items) {
+			if (item == existing_item) {
+				PX4_ERR("work item already registered, skipping");
```

Comment:
This needs to go after `ATOMIC_LEAVE`.

Also such low-level methods should return an appropriate error code instead of using printf.